horace_core/sqw/@SQWDnDBase/IX_dataset_3d.m

```diff
@@ -0,0 +1,59 @@
+function wout = IX_dataset_3d (w)
+% Convert 3D sqw object into IX_dataset_3d
+%
+%   >> wout = IX_dataset_3d (w)
+
+% R.A. Ewings, 14/10/08.
+
+
+% Check input
+if isempty(w)
+    error('sqw object is an empty array')
+end
+for i=1:numel(w)
+    if dimensions(w(i))~=3
+        if numel(w)==1
+            error('sqw object is not three dimensional')
+        else
+            error('Not all elements in the array of sqw objects are three dimensional')
+        end
+    end
+end
+
+% Fill output
+wout=IX_dataset_3d;
+if numel(w)>1, wout(numel(w))=wout; end  % allocate array
+
+for i=1:numel(w)
```

Comment:
references to `w(i).data` should be updated to `w(i).data_` to ensure the functions work with DnD objects